composer/trainer/trainer.py

```diff
@@ -1022,9 +1100,157 @@ def saved_checkpoints(self) -> List[Tuple[Timestamp, List[pathlib.Path]]]:
             return []
         return self._checkpoint_saver.saved_checkpoints
 
-    def fit(self):
-        """Train and evaluate the model on the provided data."""
-        # Print any exception, so it can be caputred by any callbacks or loggers (e.g. WandB, FileLogger)
+    def fit(
+        self,
+        *,
+        # Train Dataloader
+        train_dataloader: Optional[Union[Iterable, DataSpec, Dict[str, Any]]] = None,
+        train_dataloader_label: str = "train",
+        train_subset_num_batches: Optional[int] = None,
+        compute_training_metrics: Optional[bool] = None,
+
+        # Timing
+        max_duration: Optional[Union[int, str, Time[int]]] = None,
+        reset_timer: bool = False,
+
+        # Schedulers
+        schedulers: Optional[Union[ComposerScheduler, PyTorchScheduler, Sequence[Union[ComposerScheduler,
+                                                                                       PyTorchScheduler]]]] = None,
+        scale_schedule_ratio: float = 1.0,
+        step_schedulers_every_batch: Optional[bool] = None,
+
+        # Evaluation
+        eval_dataloader: Optional[Union[Iterable, DataSpec, Evaluator, Sequence[Evaluator]]] = None,
+        eval_subset_num_batches: int = -1,
+        eval_interval: Union[int, str, Time, Callable[[State, Event], bool]] = 1,
+
+        # Numerics
+        grad_accum: Optional[Union[int, str]] = None,
+        precision: Optional[Union[str, Precision]] = None,
+
+        # Grad Clipping
+        grad_clip_norm: Optional[float] = None,
+    ):
+        """Train the model.
+
+        .. note::
+
+            All arguments to :meth:`.fit` are optional. Any values specified here will override
+            what was provided when constructing the :class:`.Trainer`.
+
+        Args:
+            train_dataloader (Iterable | DataSpec | Dict[str, Any], optional): See :class:`.Trainer`.
+            train_dataloader_label (str, optional): See :class:`.Trainer`.
+            train_subset_num_batches (int, optional): See :class:`.Trainer`.
+            compute_training_metrics (bool, optional): See :class:`.Trainer`.
+            max_duration (Time[int] | str | int, optional): See :class:`.Trainer`.
```

Comment:
Good point; I ran into this when debugging the tests. I think it could make sense to raise an exception if `state.timer == max_duration`? Given that max duration could be specified in `__init__`, it should not need to be specified here for the first call to `fit()`.

Reply:
The new max_duration may be in different units than the existing max_duration, so it might not necessarily work to add them together.

Reply:
Refactored as described above. Added in additional tests.